drivers/network/netio/netio.c

```diff
@@ -0,0 +1,896 @@
+    if (BufferData == NULL)
+    {
+        DbgPrint("Error mapping MDL\n");
+        Irp->IoStatus.Status = STATUS_INSUFFICIENT_RESOURCES;
+        return STATUS_INSUFFICIENT_RESOURCES;
+    }
+
+    /* Some drivers (like WinDRBD) do not set this.
+     * But ReactOS seems to require this (else BSOD on completion).
+     */
+
+    if (Irp->Tail.Overlay.Thread == NULL)
+    {
+        Irp->Tail.Overlay.Thread = PsGetCurrentThread();
+    }
+    IoMarkIrpPending(Irp);
+    SocketGet(s);
+    /* TODO: protect by spinlock ... */
+    /* TODO: still needed? */
+    InsertTailList(&s->PendingUserIrps, &Irp->Tail.Overlay.ListEntry);
+
+    /* This will create a tdiIrp: */
+    status =
+        TdiSendDatagram(&tdiIrp, s->LocalAddressFile, ((char *)BufferData) + Buffer->Offset, Buffer->Length,
+                        TargetConnectionInfo, NetioComplete, nc);
+
+    /* TODO: check this again..I would assume if the TDI helper function
+       returns any error, the completion function is not called.
+     */
+    if (!NT_SUCCESS(status))
+    {
+        SocketPut(s);
+        ExFreePoolWithTag(nc, 'NEIO');
+    }
+    return status;
+}
+
+static NTSTATUS WSKAPI
+WskReceiveFrom(_In_ PWSK_SOCKET Socket,
+               _In_ PWSK_BUF Buffer,
+               _Reserved_ ULONG Flags,
+               _Out_opt_ PSOCKADDR RemoteAddress,
+               _Inout_ PULONG ControlLength,
+               _Out_writes_bytes_opt_(*ControlLength) PCMSGHDR ControlInfo,
+               _Out_opt_ PULONG ControlFlags, _Inout_ PIRP Irp)
+{
+    DbgPrint("WskReceiveFrom: Not implemented\n");
+    return STATUS_NOT_IMPLEMENTED;
+}
+
+static NTSTATUS WSKAPI
+WskReleaseUdp(_In_ PWSK_SOCKET Socket, _In_ PWSK_DATAGRAM_INDICATION DatagramIndication)
+{
+    DbgPrint("WskReleaseUdp: Not implemented\n");
+    return STATUS_NOT_IMPLEMENTED;
+}
+
+static NTSTATUS WSKAPI
+WskReleaseTcp(_In_ PWSK_SOCKET Socket, _In_ PWSK_DATA_INDICATION DataIndication)
+{
+    DbgPrint("WskReleaseTcp: Not implemented\n");
+    return STATUS_NOT_IMPLEMENTED;
+}
+
+static NTSTATUS WSKAPI
+WskGetLocalAddress(_In_ PWSK_SOCKET Socket, _Out_ PSOCKADDR LocalAddress, _Inout_ PIRP Irp)
+{
+    DbgPrint("WskGetLocalAddress: Not implemented\n");
+    return STATUS_NOT_IMPLEMENTED;
+}
+
+static NTSTATUS WSKAPI
+WskGetRemoteAddress(_In_ PWSK_SOCKET Socket, _Out_ PSOCKADDR RemoteAddress, _Inout_ PIRP Irp)
+{
+    DbgPrint("WskGetRemoteAddress: Not implemented\n");
+    return STATUS_NOT_IMPLEMENTED;
+}
+
+static NTSTATUS WSKAPI
+WskSocketConnect(PWSK_CLIENT Client,
+                 USHORT SocketType,
+                 ULONG Protocol,
+                 PSOCKADDR LocalAddress,
+                 PSOCKADDR RemoteAddress,
+                 ULONG Flags,
+                 PVOID SocketContext,
+                 const WSK_CLIENT_CONNECTION_DISPATCH * Dispatch,
+                 PEPROCESS OwningProcess,
+                 PETHREAD OwningThread, PSECURITY_DESCRIPTOR SecurityDescriptor, PIRP Irp)
```

Comment:
```suggestion
WskSocketConnect(
    PWSK_CLIENT Client,
    USHORT SocketType,
    ULONG Protocol,
    PSOCKADDR LocalAddress,
    PSOCKADDR RemoteAddress,
    ULONG Flags,
    PVOID SocketContext,
    const WSK_CLIENT_CONNECTION_DISPATCH * Dispatch,
    PEPROCESS OwningProcess,
    PETHREAD OwningThread, PSECURITY_DESCRIPTOR SecurityDescriptor, PIRP Irp)
```
do not depend on the func-names length with indent.